### ISSUE TITLE
Flang does not allow expression as argument for norm2

### ISSUE THREAD

**Issue by annop-w** [flang-compiler/flang#1361]
Example:
```
program norm2_error

  implicit none
  real, dimension(3) :: a,b,c

  a = 2.0
  b = 1.0

#ifdef WORKS
  c = a-b
  print *, norm2(c)
#else
  print *, norm2(a-b)
#endif

end program norm2_error
```
The error message is

> F90-S-0000-Internal compiler error. transform_call:Array Expression can't be here      32  (norm2.F90: 13)
  0 inform,   0 warnings,   1 severes, 0 fatal for norm2_error

**Comment by kiranchandramohan** [comment]
@shivaramaarao Do you have fix for this?

**Comment by pawosm-arm** [comment]
This is the same issue as I described here: https://github.com/flang-compiler/flang/issues/893#issuecomment-939326137

**Comment by shivaramaarao** [comment]
> program norm2_error
> 
>   implicit none
>   real, dimension(3) :: a,b,c
> 
>   a = 2.0
>   b = 1.0
> 
> #ifdef WORKS
>   c = a-b
>   print *, norm2(c)
> #else
>   print *, norm2(a-b)
> #endif
> 
> end program norm2_error



> @shivaramaarao Do you have fix for this?

Yes. Our downstream compiler has fix for this. I will create a pull request for this.

**Comment by pawosm-arm** [comment]
Fixed by PR #1365
